config/config.go

```diff
@@ -64,10 +64,23 @@ func New(cfg *common.Config) (Config, error) {
 }
 
 func (c *Config) Update(cfg *common.Config) error {
+	// Overwrite the parts of the config received in the update body.
 	if err := cfg.Unpack(&c); err != nil {
 		return err
 	}
 
+	// Shorter slices in the incoming config won't resize the corresponding slice
+	// in the current config, leaving undesirable values in the tail of the slice.
+	// This needs to be handled.
+	uc, err := New(cfg)
+	if err != nil {
+		return err
+	}
+
+	for i := range c.Streams {
+		c.Streams[i].DataYaml.ActivatedRules.CISK8S = uc.Streams[i].DataYaml.ActivatedRules.CISK8S
```

Comment:
I would add a test to cover that.
Maybe try to add a new stream as a final state in your tests?

Reply:
Changed the implementation.